### PR TITLE
SDIT-2002 Add extension helper for publish topic

### PR DESCRIPTION
**Pull request by andymarke**
This will be default retry 3 times with a small backoff

### CODE REVIEW

File: hmpps-sqs-spring-boot-autoconfigure/src/main/kotlin/uk/gov/justice/hmpps/sqs/HmppsTopic.kt [hunk 1]
```diff
@@ -1,9 +1,55 @@
...
+ */
+fun HmppsTopic.publish(
+  eventType: String,
+  event: String,
+  attributes: Map<String, MessageAttributeValue> = mapOf(
+    "eventType" to MessageAttributeValue.builder().dataType("String").stringValue(eventType).build(),
+  ),
+  retryPolicy: RetryPolicy = DEFAULT_RETRY_POLICY,
+  backOffPolicy: BackOffPolicy = DEFAULT_BACKOFF_POLICY,
+): PublishResponse {
```
Comment (mikehalmamoj) [inline hmpps-sqs-spring-boot-autoconfigure/src/main/kotlin/uk/gov/justice/hmpps/sqs/HmppsTopic.kt:42]: Shoudl we have another version of this where clients can pass in their own PublishRequest? Or is everyone in HMPPS publishing in exactly the same way?

Comment (andymarke) [reply hmpps-sqs-spring-boot-autoconfigure/src/main/kotlin/uk/gov/justice/hmpps/sqs/HmppsTopic.kt:42]: I can't see anything in the request that any hmpps service would ever set e.g 
private final String phoneNumber;
    private final String message;
    private final String subject;
    
    I think we will only ever set message

Comment (andymarke) [reply hmpps-sqs-spring-boot-autoconfigure/src/main/kotlin/uk/gov/justice/hmpps/sqs/HmppsTopic.kt:42]: I *very* much doubt anyone is using any of that

**Review by petergphillips — APPROVED** [review]
LGTM

**Pull request merged**